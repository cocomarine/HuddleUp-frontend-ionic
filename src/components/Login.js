import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import axios from "axios";
import { useAuthContext } from "../hooks/useAuthContext";
import Alert from "./Alert";

import "../styles/common/titles.css";
import "../styles/common/page.css";
import "../styles/common/buttons.css";
import "../styles/login.css";

const Login = () => {
  const [alert, setAlert] = useState({ message: "", isSuccess: false });

  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const { dispatch } = useAuthContext();

  const navigate = useNavigate();

  const changeLocation = (redirect) => {
    navigate(redirect, { replace: true });
    window.location.reload();
  };

  const handleSignin = async (e) => {
    e.preventDefault();

    setAlert({ message: "", isSuccess: false });

    axios
      .post("http://localhost:4000/auth/signin", { email, password })
      .then((res) => {
        setAlert({
          message: `${res.data.message}`,
          isSuccess: true,
        });
        localStorage.setItem("user", JSON.stringify(res.data));
        dispatch({ type: "LOGIN", payload: res.data });

        changeLocation("/myevents");
        return res.data;
      })
      .catch((err) => {
        setAlert({
          message: `${err.response.data.message}`,
          isSuccess: false,
        });
      });
  };

  return (
    <div className="loginPage">
      <h1 className="titleSignin"> Login</h1>
      <div className="loginForm">
        <form onSubmit={handleSignin}>
          <div className="labels">
            <label className="email" htmlFor="email">
              Email{" "}
            </label>
            <input
              type="email"
              id="email"
              placeholder="youremail@email.com"
              onChange={(e) => setEmail(e.target.value)}
              value={email}
              required
            ></input>
            <label htmlFor="password">Password </label>
            <input
              type="password"
              id="password"
              placeholder="*******"
              onChange={(e) => setPassword(e.target.value)}
              value={password}
              required
            ></input>
          </div>
          <div className="login">
            <button type="submit">Log in</button>
          </div>
        </form>
      </div>
      <div className="createNewAccount">
        <p>
          Don't have an account?{" "}
          <span
            className="signUpHere"
            onClick={() => {
              navigate("/signUp");
            }}
          >
            Sign Up Here
          </span>
        </p>
      </div>
    </div>
  );
};

export default Login;
