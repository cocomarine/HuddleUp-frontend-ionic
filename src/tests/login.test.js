import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Login from "../components/Login";
import { AuthContextProvider } from "../contexts/AuthContext";

const validProps = {
  onClick: jest.fn(),
  onSubmit: jest.fn(),
};

beforeEach(() => {
  render(
    <BrowserRouter>
      <AuthContextProvider>
        <Login onClick={validProps.onClick} onSubmit={validProps.onSubmit} />
      </AuthContextProvider>
    </BrowserRouter>
  );
});

describe("Login", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <AuthContextProvider>
        <Login />
      </AuthContextProvider>
    </BrowserRouter>
  );
  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the button Log In correctly", () => {
    const button = screen.getAllByRole("button");

    const loginButton = screen.getByText(/Log in/i);
    expect(loginButton).toBeInstanceOf(HTMLButtonElement);
  });

  it("redirects you upon logging in being successful", () => {
    const loginButton = screen.getByText(/Log in/i);
    fireEvent.click(loginButton);
  });
});
