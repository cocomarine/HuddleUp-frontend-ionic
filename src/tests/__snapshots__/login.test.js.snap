// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<DocumentFragment>
  <div
    class="loginPage page"
  >
    <h3
      class="titleSignin page-title"
    >
       Login
    </h3>
    <div
      class="loginForm"
    >
      <form>
        <div
          class="labels"
        >
          <label
            class="login-email subtitle"
            for="email"
          >
            Email
          </label>
          <input
            id="email"
            placeholder="youremail@email.com"
            required=""
            type="email"
            value=""
          />
          <label
            class="login-password subtitle"
            for="password"
          >
            Password 
          </label>
          <input
            id="password"
            placeholder="*******"
            required=""
            type="password"
            value=""
          />
        </div>
        <div
          class="login"
        >
          <button
            class="login-btn link-button"
            type="submit"
          >
            Log in
          </button>
        </div>
      </form>
    </div>
    <div
      class="createNewAccount heading1"
    >
      <p>
        Don't have an account? 
        <span
          class="signUpHere"
        >
          Sign Up Here
        </span>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
